### ISSUE TITLE
revise requirements/instructions for git usage for remaining problem sets

### ISSUE THREAD

**Issue by ozanj** [anyone-can-cook/rclass2#30]
Hello @augias @joycehnguy @xochilthlopez !

@augias -- this task falls on you.

I would like you to revise the requirements/instructions around using git for the remaining problem sets

for now, just focus on PS6, which is the only place where we lay out requirements and guidelines in detail.

in current requirements: we require that students do all work in dev branches, that they push their individual dev branch to the remote (so TAs can see them), and that they submit the PS to the main branch.

revised requirements I want: we require that students work in dev branches and require that they submit PS to main branch, but we do not require that they push their dev branch to the remote.

why do I want this change?

- in my own git usage, I rarely push dev branches to remote, and I usually make very short lived dev branches, that I delete as soon as some minor task is complete.
- the current requirement sort of makes them have a single dev branch that they keep around and they must constantly merge to main in order to incorporate changes to the main made by others. 

@augias - can you implement this change to the instructions for P6?

Also, try to reduce complexity/length of the "using git and github" part of the instructions for PS6

@xochilthlopez or @joycehnguy -- can one of you "assign" yourself for reviewing the revised instructions created by @augias ?

thanks team!

**Comment by augias** [comment]
Gotcha, will revise this afternoon

**Comment by ozanj** [comment]
thx @augias!

**Comment by augias** [comment]
OK, clarifying question, @ozanj 

New instructions will tell students to merge from dev_name to main. Cool. That script has everything we're actually grading and the script and data folders carry individual names. What about the Rmd? Is this necessary anymore?

Afaict, students won't actually be doing any work in markdown this problem set. The TAs can fill in their own ps6 markdown file with grades and comments, but the work is in their scripts. 

Soooo... I am proposing we tell them in the beginning of ps6 (ie the overview): "Don't fill out your Rmd file, all your answers and work will be graded based on your ps6.R script."

And at the end of the assignment in the section "Knit to pdf and submit problem set" I can change it to "To submit your work, merge dev_lastname_firstname into main and push your changes to remote."

Pretty minimal there. 

Edit: PS6 already instructs students to turn hw in using the main branch. All I need to do is add instructions at the end of the document and tighten up the git and github section?

**Comment by ozanj** [comment]
Hi @augias 

this seems good to me. 

though for some remaining problem sets, they might have to write answers in the .Rmd. I forget.

**Comment by augias** [comment]
Perfect, I'll keep that in mind and look ahead right now. 
@joycehnguy @xochilthlopez please check my first draft of edits and changes!
[PDF](https://ucla.box.com/s/06docgj8byoxivgif3rbuotoipc6b54i)
[Rmd](https://ucla.box.com/s/lltc4jbq34dmdwslc64cu7pfvhi9n1e2)

I've edited the "using git and github" section to be a little more minimal. There were some really detailed instructions I feel we can assume they can figure out on their own now.

Also edited the final "submit problem set" section.

**Comment by joycehnguy** [comment]
Yes - I just texted Fernando about if we are doing only R scripts, the only thing is the posting of github links. So we either will 1) push both Rmd and R scripts which students are used to or 2) just add the instructions to add the github links to the R script

**Comment by ozanj** [comment]
Thx yall,

Generally speaking there might be other files that are part of repo for a problem set, like folders plots datasets. So students may push those too in some problem sets too. But I forget. I'm at a red light!

**Comment by augias** [comment]
Alright, I added a note in the github comments and issues section right at the start letting students know they can put the url links to their posts on github issues in their R scripts. I'll do the same for all subsequent weeks if those psets don't require use of the Rmd file
